airbyte-integrations/connectors/source-optimizely-campaigns/source_optimizely_campaigns/source.py

```diff
@@ -0,0 +1,371 @@
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        if self.client_id:
+            return {"Optimizely-Account": self.client_id}
+
+        return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+
+        yield from response_json.get("elements", [])
+
+
+class IncrementalOptimizelyStream(OptimizelyStream, ABC):
+    cursor_field = "modified"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+
+        return {self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, ""))}
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["modified[gte]"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class RecipientLists(IncrementalOptimizelyStream):
+    primary_key = ["id"]
+    cursor_field = "modified"
+
+    def path(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> str:
+        return "/recipientlists"
+
+
+class Recipients(IncrementalOptimizelyStream):
+    primary_key = ["id", "recipient_list"]
+    cursor_field = "modified"
+
+    def path(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> str:
+        recipient_list = stream_slice["list_id"]
+        return f"/recipients/{recipient_list}"
+
+    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+        attribute_names_stream = AttributeNames(authenticator=self.authenticator, client_id=self.client_id)
+        for attribute_name in attribute_names_stream.read_records(sync_mode=SyncMode.full_refresh):
+            attribute_name_string = ",".join(attribute_name["attributeNames"])
+            slices = {"list_id": attribute_name["recipient_list"], "attributes": attribute_name_string}
+            yield from super().read_records(stream_slice=slices, **kwargs)
+
+    def request_params(self, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_slice=stream_slice, **kwargs)
+        params = {"limit": self.API_RECIPIENT_LIMIT, "attributeNames": stream_slice["attributes"]}
```

Comment:
is this super() call necessary given you're overwriting the params variable afterwards?